### PR TITLE
AV-157454 Handle AviInfraSetting failures

### PR DESCRIPTION
**Pull request by abhishekbsingh**

### CODE REVIEW

**Comment by abhishekbsingh** [comment]
build ako

**Comment by abhishekbsingh** [comment]
build ako

**Comment by abhishekbsingh** [comment]
build ako

File: internal/nodes/ingress_model_rel.go [hunk 1]
```diff
@@ -1182,6 +1182,9 @@ func infraSettingNSToServices(infraSettingName, key string) ([]string, bool) {
 			return allServices, false
 		}
 		for _, svc := range services {
+			if svc.Spec.Type != "LoadBalancer" {
```
Comment (akshayhavile) [inline internal/nodes/ingress_model_rel.go:1185]: please use function `isServiceLBType`

**Review by akshayhavile — COMMENTED** [review]
@abhishekbsingh: There are UT failures for your code changes. Please check those.

**Comment by abhishekbsingh** [comment]
> @abhishekbsingh: There are UT failures for your code changes. Please check those.

The failing UT actually passed locally, checking further

File: internal/nodes/ingress_model_rel.go (hunk 1 above)
Comment (abhishekbsingh) [reply internal/nodes/ingress_model_rel.go:1185]: that function will not be available to other packages (need Upper Camel case)

**Comment by abhishekbsingh** [comment]
build ako

**Comment by abhishekbsingh** [comment]
build ako

**Comment by abhishekbsingh** [comment]
build ako

**Comment by abhishekbsingh** [comment]
build ako

**Comment by abhishekbsingh** [comment]
build ako

**Comment by abhishekbsingh** [comment]
build ako

File: internal/nodes/dequeue_ingestion.go [hunk 2]
```diff
@@ -263,9 +263,11 @@ func DequeueIngestion(key string, fullsync bool) {
 				} else {
 					aviModelGraph := NewAviObjectGraph()
 					aviModelGraph.BuildAdvancedL4Graph(namespace, gwName, key, false)
-					ok := saveAviModel(modelName, aviModelGraph, key)
-					if ok && len(aviModelGraph.GetOrderedNodes()) != 0 && !fullsync {
-						PublishKeyToRestLayer(modelName, key, sharedQueue)
+					if len(aviModelGraph.GetOrderedNodes()) > 0 {
+						ok := saveAviModel(modelName, aviModelGraph, key)
+						if ok && len(aviModelGraph.GetOrderedNodes()) != 0 && !fullsync {
```
Comment (swathinsankaran) [inline internal/nodes/dequeue_ingestion.go:268]: `len(aviModelGraph.GetOrderedNodes()) != 0` is already taken care in L266 it seems.

File: internal/nodes/dequeue_ingestion.go [hunk 3]
```diff
@@ -644,9 +646,11 @@ func handleL4SharedVipService(namespacedVipKey, key string, fullsync bool) {
 		aviModelGraph := NewAviObjectGraph()
 		vipKey := strings.Split(namespacedVipKey, "/")[1]
 		aviModelGraph.BuildAdvancedL4Graph(namespace, vipKey, key, true)
-		ok := saveAviModel(modelName, aviModelGraph, key)
-		if ok && len(aviModelGraph.GetOrderedNodes()) != 0 && !fullsync {
-			PublishKeyToRestLayer(modelName, key, sharedQueue)
+		if len(aviModelGraph.GetOrderedNodes()) > 0 {
+			ok := saveAviModel(modelName, aviModelGraph, key)
+			if ok && len(aviModelGraph.GetOrderedNodes()) != 0 && !fullsync {
```
Comment (swathinsankaran) [inline internal/nodes/dequeue_ingestion.go:651]: `len(aviModelGraph.GetOrderedNodes()) != 0` is already taken care in L649 it seems.

**Review by akshayhavile — APPROVED** [review]

**Comment by abhishekbsingh** [comment]
build ako

**Comment by abhishekbsingh** [comment]
build ako

**Comment by abhishekbsingh** [comment]
build ako

**Comment by abhishekbsingh** [comment]
build ako

**Review by swathinsankaran — APPROVED** [review]
LGTM

**Review by akshayhavile — APPROVED** [review]

**Pull request merged**